### PR TITLE
Remove WRITE usages of compatDetails on ContainerRuntime GC ops

### PR DESCRIPTION
**Pull request by MarioJGMsoft**
## Description

Fixes: [AB#19718](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/19718)

Follow-on to [#17024](https://dev.azure.com/fluidframework/internal/_workitems/edit/17024/)

We need to look at Kusto first to confirm our theory that the code to read both the GC op types ("Sweep" and "TombstoneLoaded") has sufficiently saturated.
"Sweep" was [added](https://github.com/microsoft/FluidFramework/commit/b3b6cd4e09659521e11cf57d5663c48d2ce7e9c1) in 2.0.0-internal.7.4.0
"TombstoneLoaded" was [added](https://github.com/microsoft/FluidFramework/commit/dd9ff1665bfb5f10ff59e02ce4caa2b674f4de47) in 2.0.0-rc.2.0.0
We do see sessions on RC.1, but I believe they're all Whiteboard, which doesn't see any "TombstoneLoaded" events which trigger those ops.

We can write a query to confirm with high confidence, then proceed.

## Reviewer Guidance

The goal of this PR is to remove all uses of compatDetails that are no longer necessary, please let me know if I missed any or if there are some that can't be removed yet.

### CODE REVIEW

File: packages/runtime/container-runtime/src/test/gc/garbageCollection.spec.ts [hunk 1]
```diff
@@ -2190,43 +2189,24 @@ describe("Garbage Collection Tests", () => {
 			});
 		});
```
Comment (markfields) [inline packages/runtime/container-runtime/src/test/gc/garbageCollection.spec.ts:2190]: You can delete the whole describe block "Future GC op type compatibility"

Comment (markfields) [reply packages/runtime/container-runtime/src/test/gc/garbageCollection.spec.ts:2190]: Ok I guess keeping the test "process remote op with unrecognized type" somewhere would be good, but it doesn't need to be under a describe block.

File: packages/runtime/container-runtime/src/test/pendingStateManager.spec.ts [hunk 2]
```diff
@@ -601,11 +600,9 @@ describe("Pending State Manager", () => {
 		describe("Future op compat behavior", () => {
```
Comment (markfields) [inline packages/runtime/container-runtime/src/test/pendingStateManager.spec.ts:600]: Just delete this test

File: packages/test/test-end-to-end-tests/src/test/batching.spec.ts [hunk 3]
```diff
@@ -183,7 +183,6 @@ describeCompat("Flushing ops", "NoCompat", (getTestObjectProvider, apis) => {
 				type: "TombstoneLoaded",
```
Comment (markfields) [inline packages/test/test-end-to-end-tests/src/test/batching.spec.ts:183]: Just delete this whole test

**Review by markfields — COMMENTED** [review]
Good start - Do a find-all for compatDetails, there's more to delete!

File: packages/test/test-end-to-end-tests/src/test/batching.spec.ts (hunk 3 above)
Comment (MarioJGMsoft) [reply packages/test/test-end-to-end-tests/src/test/batching.spec.ts]: Done

File: packages/runtime/container-runtime/src/test/pendingStateManager.spec.ts (hunk 2 above)
Comment (MarioJGMsoft) [reply packages/runtime/container-runtime/src/test/pendingStateManager.spec.ts:607]: Done

File: packages/runtime/container-runtime/src/test/gc/garbageCollection.spec.ts (hunk 1 above)
Comment (MarioJGMsoft) [reply packages/runtime/container-runtime/src/test/gc/garbageCollection.spec.ts]: I'm working on making the test function properly

File: packages/runtime/container-runtime/src/gc/garbageCollection.ts [hunk 4]
```diff
@@ -715,7 +715,6 @@ export class GarbageCollector implements IGarbageCollector {
 			const containerGCMessage: ContainerRuntimeGCMessage = {
```
Comment (markfields) [inline packages/runtime/container-runtime/src/gc/garbageCollection.ts:715]: Just noticed the 3-line comment above this line - should be deleted too.

File: packages/runtime/container-runtime/src/gc/garbageCollection.ts [hunk 5]
```diff
@@ -1044,7 +1043,6 @@ export class GarbageCollector implements IGarbageCollector {
 				type: GarbageCollectionMessageType.TombstoneLoaded,
```
Comment (markfields) [inline packages/runtime/container-runtime/src/gc/garbageCollection.ts:1043]: And the comment above this too.

File: packages/runtime/container-runtime/src/test/gc/garbageCollection.spec.ts [hunk 6]
```diff
@@ -2181,61 +2180,21 @@ describe("Garbage Collection Tests", () => {
...
-		});
+		assert.throws(
+			() =>
+				garbageCollector.processMessages(
+					[gcMessageFromFuture as unknown as GarbageCollectionMessage],
+					Date.now(),
+					false /* local */,
+				),
+			(error: IErrorBase) => error.errorType === ContainerErrorTypes.dataProcessingError,
+			"Garbage collection message of unknown type FROM_THE_FUTURE",
```
Comment (markfields) [inline packages/runtime/container-runtime/src/test/gc/garbageCollection.spec.ts:2197]: Noticed a typo from before - `FROM_THE_FUTURE` should be `FUTURE_MESSAGE`, I think?

File: packages/runtime/container-runtime/src/test/containerRuntime.spec.ts [hunk 7]
```diff
@@ -1129,159 +1120,6 @@ describe("Runtime", () => {
 			);
 		});
 
-		describe("[DEPRECATED] Future op type compatibility", () => {
```
Comment (markfields) [inline packages/runtime/container-runtime/src/test/containerRuntime.spec.ts:1132]: Sorry... I think many of these tests are good to keep around.  They are illustrating that things fail with an unknown type.  So I'd call this section "Unrecognized types not supported", and make sure the titles are correct for each of the tests that assert that something throws (e.g. the Resubmit one still says something about being ignored).

Of course _do_ delete the first test "can submit op compat behavior"

Comment (markfields) [reply packages/runtime/container-runtime/src/test/containerRuntime.spec.ts:1132]: And get rid of the compatDetails in each test case too, of course. That's what tripped me up about saying to delete them I think.

**Review by markfields — APPROVED** [review]
LGTM, once you address the last few comments! Thanks for knocking this out!

File: packages/runtime/container-runtime/src/test/gc/garbageCollection.spec.ts (hunk 6 above)
Comment (MarioJGMsoft) [reply packages/runtime/container-runtime/src/test/gc/garbageCollection.spec.ts]: done

File: packages/runtime/container-runtime/src/gc/garbageCollection.ts (hunk 4 above)
Comment (MarioJGMsoft) [reply packages/runtime/container-runtime/src/gc/garbageCollection.ts:748]: done

File: packages/runtime/container-runtime/src/gc/garbageCollection.ts (hunk 5 above)
Comment (MarioJGMsoft) [reply packages/runtime/container-runtime/src/gc/garbageCollection.ts:1075]: done

File: packages/runtime/container-runtime/src/test/containerRuntime.spec.ts (hunk 7 above)
Comment (MarioJGMsoft) [reply packages/runtime/container-runtime/src/test/containerRuntime.spec.ts:1132]: The only tests that remained are Op with unrecognized type is ignored by resubmit and process remote op with unrecognized type. Those are the only ones that I found necessary but let me know if you think we should keep anything else.

**Pull request merged**